src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -2015,6 +2018,30 @@ Node TheoryStringsRewriter::rewriteIndexof( Node node ) {
       return returnRewrite(node, negone, "idof-nctn");
     }
   }
+  else
+  {
+    Node new_len = node[2];
+    std::vector<Node> nr;
+    if (stripSymbolicLength(children0, nr, 1, new_len))
+    {
+      // Normalize the string before the start index.
+      //
+      // For example:
+      // str.indexof(str.++("ABCD", x), y, 3) --->
+      // str.indexof(str.++("AAAD", x), y, 3)
+      Node normNr = lengthPreserveRewrite(mkConcat(kind::STRING_CONCAT, nr));
```

Comment:
Store mkConcat(kind::STRING_CONCAT, nr) to a variable and use here and in the line below.